cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectApiClientTest.java

```diff
@@ -2,51 +2,90 @@
+        "https://api.cristin-test.uio.no/v2/projects?lang=nb&page=1&per_page=5&title=reindeer";
     private static final String CRISTIN_API_GRANT_ID_SEARCH_EXAMPLE_URI =
-        "https://api.cristin-test.uio.no/v2/projects?per_page=5&project_code=1234567&page=1&lang=nb";
-
-    final QueryCristinProjectApiClient cristinApiClient = new QueryCristinProjectApiClient();
+        "https://api.cristin-test.uio.no/v2/projects?lang=nb&page=1&per_page=5&project_code=1234567";
 
     @Test
     void getsCristinUriWithTitleParamWhenCallingUriBuilderWithTitleQueryRequested() throws Exception {
-        Map<String, String> params = Map.of(
-            QUERY, RANDOM_TITLE,
-            LANGUAGE, LANGUAGE_NB,
-            PAGE, FIRST_PAGE,
-            NUMBER_OF_RESULTS, DEFAULT_NUMBER_OF_RESULTS);
-        URI expectedUri = new URI(QUERY_CRISTIN_PROJECTS_EXAMPLE_URI);
-
-        assertEquals(expectedUri, cristinApiClient.generateQueryProjectsUrl(params, QUERY_USING_TITLE));
+        final var cristinQuery =
+            CristinQuery.builder()
+                .withQuery(RANDOM_TITLE)
+                .withLanguage(LANGUAGE_NB)
+                .withItemsFromPage(FIRST_PAGE)
+                .withItemsPerPage(DEFAULT_NUMBER_OF_RESULTS)
+                .validate()
+                .build();
+        final var expectedUri = new URI(QUERY_CRISTIN_PROJECTS_EXAMPLE_URI);
+        final var actualUri1 = generateCristinQueryProjectsUrl(cristinQuery.toNvaParameters());
+        final var actualUri2 = cristinQuery.toURI();
+        assertEquals(expectedUri, actualUri1);
+        assertEquals(expectedUri, actualUri2);
+    }
+
+    @Test
+    void getsCristinUriWithNoQueryParamWhenCallingUriBuilderWithTitleQueryRequested() throws Exception {
+        final var params =
+            CristinQuery.builder()
+                .withTitle(RANDOM_TITLE)
+                .withLanguage(LANGUAGE_NB)
+                .withItemsFromPage(FIRST_PAGE)
+                .withItemsPerPage(DEFAULT_NUMBER_OF_RESULTS)
+                .validate()
+                .build()
+                .toParameters();
+        final var expectedUri = new URI(QUERY_CRISTIN_PROJECTS_EXAMPLE_URI);
+        final var actualUri = generateCristinQueryProjectsUrl(params);
+        assertEquals(expectedUri, actualUri);
     }
 
     @Test
     void getsCristinUriWithProjectCodeParamWhenCallingUriBuilderWithGrantIdQueryRequested() throws Exception {
-        Map<String, String> params = Map.of(
-            QUERY, GRANT_ID_EXAMPLE,
-            LANGUAGE, LANGUAGE_NB,
-            PAGE, FIRST_PAGE,
-            NUMBER_OF_RESULTS, DEFAULT_NUMBER_OF_RESULTS);
-        URI expectedUri = new URI(CRISTIN_API_GRANT_ID_SEARCH_EXAMPLE_URI);
-
-        assertEquals(expectedUri, cristinApiClient.generateQueryProjectsUrl(params, QUERY_USING_GRANT_ID));
+        final var sourceCristinQuery =
+            CristinQuery.builder()
+                .withQuery(GRANT_ID_EXAMPLE)
```

Comment:
query or grant_id (project_code)?

Reply:
```
    public CristinQueryBuilder withQuery(String query) {
        if (nonNull(query)) {
            var key = grantOrTitleKey(query);
            cristinQuery.setValue(key, query);
```

`grantOrTitleKey()` will read content and decide if it is title or grant